src/stores/utils/contract.js

```diff
@@ -66,16 +69,42 @@ export const totalBurntCoins = async (contract) => {
   return new BN(burntCoins)
 }
 
-export const getBridgeValidators = async (bridgeValidatorContract) => {
-  let ValidatorAdded = await bridgeValidatorContract.getPastEvents('ValidatorAdded', {fromBlock: 0});
-  let ValidatorRemoved = await bridgeValidatorContract.getPastEvents('ValidatorRemoved', {fromBlock: 0});
-  let addedValidators = ValidatorAdded.map(val => {
-    return val.returnValues.validator
-  })
-  const removedValidators = ValidatorRemoved.map(val => {
-    return val.returnValues.validator
+export const getValidatorList = async (address, eth) => {
+  const v1Contract = new eth.Contract(BridgeValidatorsV1Abi, address);
+  const v1ValidatorsEvents = await getValidatorEvents(v1Contract)
+
+  const contract = new eth.Contract(BRIDGE_VALIDATORS_ABI, address);
```

Comment:
why do we need to use different ABIs here? I see that eventually we look for `ValidatorAdded` and `ValidatorRemoved` only

Reply:
We have to use different ABIs to get the past events because the events has to match the exact definition from the ABI otherwise the event isn't returned as part of the result. We have 3 different definitions for `ValidatorAdded` and `ValidatorRemoved`.
1- On v1:
```
  {
    "anonymous": false,
    "inputs": [
      {
        "indexed": false,
        "name": "validator",
        "type": "address"
      }
    ],
    "name": "ValidatorAdded",
    "type": "event"
  }
```  
2- On v2 (input is indexed):
```
{
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "name": "validator",
          "type": "address"
        }
      ],
      "name": "ValidatorAdded",
      "type": "event"
    }
```
3- On RewardableValidators (reward field added):
```
{
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "name": "validator",
          "type": "address"
        },
        {
          "indexed": false,
          "name": "reward",
          "type": "address"
        }
      ],
      "name": "ValidatorAdded",
      "type": "event"
    }
```

Same differences applies to `ValidatorRemoved`.
So for example if validators contract is on version 1, and it added a new validator. If you try to get past events using v2 abi we won't get any results

Reply:
Thanks!. I see the difference between v1 and v2 contracts. I can do nothing with this - `indexed` was added to speed up a lookup whether the corresponding validator was added/removed.
But do you think we complicated things by adding `reward` field in the new versions of contracts and broke the UI functionality existed before? Do you think it makes sense to emit a separate event to notify about reward address?

Reply:
I think it makes sense to emit an event where the validator and the reward address are linked as we did with `ValidatorAdded` on RewardableValidators.
I don't think we can benefit from emiting this information on a separate event and having two of them. So I think current implemented events are ok. 

The complicated part of this was to be able to listen and parse correctly all the variations of the events when a validator is added/removed with the limitations of web3 library.
I think with the new approach applied and mentioned here https://github.com/poanetwork/bridge-ui/pull/184#issuecomment-481661210 we found a good solution despite of having to do some low level manual parse for the events instead of letting web3 do it.

Reply:
Do you recall what was the reason to have the same event name when we have different number of the event's data fields?
I am thinking (and actually we see it here) that by introducing this we added complexity and, probably, moved our design in a wrong direction. Two different events (in fact they have different signatures produced by the bytecode) for the same information cause to increase complexity in the bridge UI, in the monitor. Imagine if tomorrow we would like to synchronize the validators sets we will need to watch two different topics by the oracle. What benifit to have the same name in Solidity code in two contracts is?

That's cool that you developed a way to handle complex situation appeared due to upgrade from v1 to v2 and we definitely will use this since we have no better choice but do we want to continue increasing complexity by handling different event topics appearing in v3? I think the RewardabeValidators contract is not compatible now due to the events signature that's why we need to increase the major version?

Reply:
>I think the RewardabeValidators contract is not compatible now due to the events signature that's why we need to increase the major version?

I agree with this. 

>Do you recall what was the reason to have the same event name when we have different number of the event's data fields?

I can't remember or find the specific reason. To avoid adding complexity, do you think we should have only one event type that works with the both validators contracts needs?

>Imagine if tomorrow we would like to synchronize the validators sets we will need to watch two different topics by the oracle.

Also methods to add new validator is different on `BridgeValidators` and `RewardableValidators`:
- `function addValidator(address _validator)`
- `function addValidator(address _validator, address _reward)`
So the complexity involves watching the two events and calling the correct methods. Maybe we should think of defining a unique API (events and methods) for both contracts

Reply:
Yes, here I am on your side completely. What about `addValidator` and `addRewardableValidator` methods and leave just one event `ValidatorAdded(address indexed validator)`? We always can the address where rewards are going to be sent by decoding the method call parameters if it is needed in the future.

Reply:
Those methods and the event sound good to me. We can also get the reward address for a validator by calling method `getValidatorRewardAddress(address _validator) public view returns (address)`. Should we create an issue for this?

Reply:
Yes, please. As soon as the issue in the contract addressed, these PR should be updated to reflect the changes in the contracts.

Reply:
Created PR https://github.com/poanetwork/poa-bridge-contracts/pull/169. With changes mentioned here https://github.com/poanetwork/bridge-ui/pull/184#discussion_r274548322 I think there are no other changes needed for this PR